### PR TITLE
feat:Startup environment log

### PR DESCRIPTION
**Pull request by marcotc**
### What does this PR do?
Adds standardized startup logging as soon as we have enough information to do so.

### Motivation
Help support get as much information as they can from app startup logs.

### Implementation
We collect logs when the first trace is attempted to be sent to the agent. This allows us to collect valuable information about the connectivity with the agent and any specific failure conditions, if any.

This information is logged only once for the application lifetime.

### CODE REVIEW

File: spec/ddtrace/diagnostics/environment_logger_spec.rb [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+
+        it do
+          # Because net/http is default gem, we use the Ruby version as the library version.
+          is_expected.to include integrations_loaded: end_with("@#{RUBY_VERSION}")
+        end
+
+        context 'with integration-specific settings' do
+          let(:options) { { service_name: 'my-http' } }
+
+          it { is_expected.to include http_analytics_enabled: false }
```
Comment (ericmustin) [inline spec/ddtrace/diagnostics/environment_logger_spec.rb:247]: small nit, the suggested property name is `integration_<integration>_analytics_enabled` per the google doc, I believe. Not sure how strict we need to be toward the doc, feel free to disregard if it's not an issue.

File: lib/ddtrace/diagnostics/environment_logger.rb [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+        !!Datadog.configuration.tracer.priority_sampling
+      end
+
+      # @return [Boolean, nil] health metrics enabled in configuration
+      def health_metrics_enabled
+        !!Datadog.configuration.diagnostics.health_metrics.enabled
+      end
+
+      # @return [Hash] environment information available at call time
+      def collect!(transport_responses)
```
Comment (ericmustin) [inline lib/ddtrace/diagnostics/environment_logger.rb:217]: is either `logs_correlation_enabled` or `log_injection_enabled` something we're able to record? Perhaps just I see that listed in the google doc but is missing here.

Comment (ericmustin) [reply lib/ddtrace/diagnostics/environment_logger.rb:217]: Also, just for posterity, we should make a `TODO` or put something in the backlog to add `profiling_enabled` here when that gets merged in

File: spec/ddtrace/diagnostics/environment_logger_spec.rb (hunk 1 above)
Comment (marcotc) [reply spec/ddtrace/diagnostics/environment_logger_spec.rb]: Nice catch, I think I was reading the wrong column in the spreadsheet when I named integration settings. Addressed!

File: lib/ddtrace/diagnostics/environment_logger.rb (hunk 2 above)
Comment (marcotc) [reply lib/ddtrace/diagnostics/environment_logger.rb:225]: We are not currently able to know if `logs_correlation_enabled`, as we don't support automatic log injection, and the snippets we suggest for users to implement it call generic functions. I'd added a placeholder for when we implement automatic log correlation injection.

I added a similar placeholder for profiling.

**Pull request merged**